### PR TITLE
New rule for SLE 15 unnecessary_firewalld_services_ports_disabled

### PR DESCRIPTION
**Pull request by rumch-se**
#### Description:

- _New rule for SLE 15 which covers CIS requirement 3.5.1.6 Ensure unnecessary services and ports are not accepted_

#### Rationale:

- At the moment CIS profile for SLE 15 does not include this requirement.  The control cannot be automated, and should be addressed manually.

### CODE REVIEW

File: linux_os/guide/system/network/network-firewalld/unnecessary_firewalld_services_ports_disabled/rule.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+prodtype: sle15
+
+title: 'Ensure unnecessary services and ports are not accepted'
+
+description: |-
+    Services and ports can be accepted or explicitly rejected or dropped by a zone.
+    For every zone, you can set a default behavior that handles incoming traffic that is not
+    further specified. Such behavior is defined by setting the target of the zone. There are three
+    options - default, ACCEPT, REJECT, and DROP.
+    - <tt>ACCEPT</tt> - you accept all incoming packets except those disabled by a specific rule.
```
Comment (Mab879) [inline linux_os/guide/system/network/network-firewalld/unnecessary_firewalld_services_ports_disabled/rule.yml:12]:
```suggestion
    - <tt>ACCEPT</tt> - accept all incoming packets except those disabled by a specific rule.
```

File: linux_os/guide/system/network/network-firewalld/unnecessary_firewalld_services_ports_disabled/rule.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    - <tt>REJECT</tt> - you disable all incoming packets except those that you have allowed in
```
Comment (Mab879) [inline linux_os/guide/system/network/network-firewalld/unnecessary_firewalld_services_ports_disabled/rule.yml:13]:
```suggestion
    - <tt>REJECT</tt> - disables all incoming packets except those that have been allowed in by
```

File: linux_os/guide/system/network/network-firewalld/unnecessary_firewalld_services_ports_disabled/rule.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+       specific rules and the source machine is informed about the rejection.
+    - <tt>DROP</tt> - you disable all incoming packets except those that you have allowed in
```
Comment (Mab879) [inline linux_os/guide/system/network/network-firewalld/unnecessary_firewalld_services_ports_disabled/rule.yml:15]:
```suggestion
    - <tt>DROP</tt> - disables all incoming packets except those that been have allowed in by
```

File: linux_os/guide/system/network/network-firewalld/unnecessary_firewalld_services_ports_disabled/rule.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+severity: medium
+
+identifiers:
+    cce@sle15: CCE-92552-9
+
+references:
+    cis@sle15: 3.5.1.6
+
+ocil_clause: 'Your system accepts all incoming packets for unnecessary services and ports'
```
Comment (Mab879) [inline linux_os/guide/system/network/network-firewalld/unnecessary_firewalld_services_ports_disabled/rule.yml:30]:
```suggestion
ocil_clause: 'the system accepts all incoming packets for unnecessary services and ports'
```

File: linux_os/guide/system/network/network-firewalld/unnecessary_firewalld_services_ports_disabled/rule.yml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+ocil: |-
+    To review and to ensure that listed services and ports follow site policy run the 
+    following command: 
+    <pre>$ sudo firewall-cmd --get-active-zones | awk '!/:/ {print $1}' | while read ZN; do
+    firewall-cmd --list-all --zone=$ZN; done</pre>
+    If your have to remove an unnecessary service run the following command:
```
Comment (Mab879) [inline linux_os/guide/system/network/network-firewalld/unnecessary_firewalld_services_ports_disabled/rule.yml:37]:
```suggestion
    To remove an unnecessary service, run the following command: 
```

File: linux_os/guide/system/network/network-firewalld/unnecessary_firewalld_services_ports_disabled/rule.yml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+ocil_clause: 'Your system accepts all incoming packets for unnecessary services and ports'
+
+ocil: |-
+    To review and to ensure that listed services and ports follow site policy run the 
+    following command: 
+    <pre>$ sudo firewall-cmd --get-active-zones | awk '!/:/ {print $1}' | while read ZN; do
+    firewall-cmd --list-all --zone=$ZN; done</pre>
+    If your have to remove an unnecessary service run the following command: 
+    <pre>$ sudo firewall-cmd --remove-service=<service></pre>
+    If you have to remove an unnecessary port run the following command:
```
Comment (Mab879) [inline linux_os/guide/system/network/network-firewalld/unnecessary_firewalld_services_ports_disabled/rule.yml:39]:
```suggestion
    To remove an unnecessary port, run the following command:
```

File: linux_os/guide/system/network/network-firewalld/unnecessary_firewalld_services_ports_disabled/rule.yml [hunk 7]
```diff
@@ -0,0 +1,40 @@
+documentation_complete: true
+
+prodtype: sle15
+
+title: 'Ensure unnecessary services and ports are not accepted'
```
Comment (Mab879) [inline linux_os/guide/system/network/network-firewalld/unnecessary_firewalld_services_ports_disabled/rule.yml:5]:
```suggestion
title: 'Ensure Unnecessary Services and Ports Are Not Accepted'
```

File: linux_os/guide/system/network/network-firewalld/unnecessary_firewalld_services_ports_disabled/rule.yml [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+description: |-
+    Services and ports can be accepted or explicitly rejected or dropped by a zone.
+    For every zone, you can set a default behavior that handles incoming traffic that is not
```
Comment (Mab879) [inline linux_os/guide/system/network/network-firewalld/unnecessary_firewalld_services_ports_disabled/rule.yml:9]:
```suggestion
    For every zone, a default behavior can be set that handles incoming traffic that is not
```

**Review by Mab879 — CHANGES_REQUESTED** [review]
Thanks for the PR. I have do a few suggestions.

**Comment by rumch-se** [comment]
Hello @Mab879 
Thank you for your feedback.
I have corrected the rule.
Have a nice day
Rumen

File: linux_os/guide/system/network/network-firewalld/unnecessary_firewalld_services_ports_disabled/rule.yml [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+severity: medium
+
+identifiers:
+    cce@sle15: CCE-92552-9
+
+references:
+    cis@sle15: 3.5.1.6
+
+ocil_clause: 'The system accepts all incoming packets for unnecessary services and ports'
```
Comment (Mab879) [inline linux_os/guide/system/network/network-firewalld/unnecessary_firewalld_services_ports_disabled/rule.yml:30]:
```suggestion
ocil_clause: 'the system accepts incoming packets for unnecessary services and ports'
```

**Review by Mab879 — CHANGES_REQUESTED** [review]
Sorry, Just one last minor change.

**Comment by rumch-se** [comment]
Hello @Mab879 
I have done the requested change.
Have a nice day
Rumen

**Pull request merged**